scripts/run_goreleaser.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+[[ -z ${DEBUG:-} ]] || set -o xtrace
+
+OS="$(uname -s | tr '[:upper:]' '[:lower:]')"
+DIR="$(cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")" && cd .. && pwd -P)"
+GITCONFIG_VOLUME=${GIT_CONFIG:-"${HOME}/.gitconfig"}
+GCLOUD_CONFIG_DIR="${CLOUDSDK_CONFIG:-${HOME}/.config/gcloud}"
+SSH_CONFIG_DIR_VOLUME="${HOME}/.ssh"
+SSH_AUTH_SOCK_ENV_VAR="/ssh-agent"
+SSH_AUTH_SOCK_VOLUME="${SSH_AUTH_SOCK}:/ssh-agent"
+
+if [[ "${BUILDKITE:-}" == "true" ]]; then
+    GITCONFIG_VOLUME="/var/lib/buildkite-agent/.gitconfig"
+    SSH_CONFIG_DIR_VOLUME="/var/lib/buildkite-agent/.ssh"
+fi
+
+# On Mac for Desktop we need to use these hardcoded values when forwarding SSH agent socket
+# Docker would 'magically' map whatever is set in $SSH_AUTH_SOCK for the user to the container ns
+if [[ "${OS}" == "darwin" ]]; then
+    SSH_AUTH_SOCK_VOLUME="/run/host-services/ssh-auth.sock:/run/host-services/ssh-auth.sock"
+    SSH_AUTH_SOCK_ENV_VAR="/run/host-services/ssh-auth.sock"
+fi
+
+DOCKER_OPTS=(
+    -w "${GO_RELEASER_WORKING_DIR}"
+    -e "GITHUB_TOKEN"
+    -e "GIT_REVISION"
+    -e "GIT_BRANCH"
+    -e "GIT_VERSION"
+    -e "BUILD_DATE"
+    -e "BUILD_TS_UNIX"
+    -e "INSTRUMENT_PACKAGE"
+    -e "GO_BUILD_LDFLAGS"
+    -e "CGO_ENABLED=0"
+    -e "SSH_AUTH_SOCK=${SSH_AUTH_SOCK_ENV_VAR}"
+    -v "${SSH_AUTH_SOCK_VOLUME}"
+    -v "${SSH_CONFIG_DIR_VOLUME}:/root/.ssh"
+    -v "${GITCONFIG_VOLUME}:/root/.gitconfig"
+    -v "${DIR}:${GO_RELEASER_WORKING_DIR}"
+)
+
+if [[ "${BUILDKITE:-}" != "true" ]]; then
+    DOCKER_OPTS+=(-v "${GCLOUD_CONFIG_DIR}:/root/.config/gcloud")
+fi
+
+echo go mod vendor
```

Comment:
can you add a note for why we call vendor. I thought it might be because we're running in a container without access to the github repos, but i think all the deps are public, and we have ssh agent in there.

I think the main reason is to take advantage of the local host cache for deps vs downloading them from scratch in a docker container?